src/dtos/dataset-dto.ts

```diff
@@ -350,7 +350,7 @@ export class DatasetDTO {
     static async fromDatasetWithShallowDimensionsAndRevisions(dataset: Dataset): Promise<DatasetDTO> {
         const dto = new DatasetDTO();
         dto.id = dataset.id;
-        dto.creation_date = dataset.creation_date.toISOString();
+        dto.creation_date = dataset.creationDate.toISOString();
```

Comment:
If you're ok with it I'll probably do another pass to rename any date columns with `xxx_at` e.g. 
 * created_at
 * updated_at
 * publish_at
 * approved_at
etc